src/utils/crypto.ts

```diff
@@ -39,3 +45,27 @@ export const createCryptoContextFor = (keytag: string): CryptoContext => {
     },
   };
 };
+
+/**
+ * Executes the input function injecting an ephemeral crypto context.
+ * An ephemeral crypto context is a context which is bound to a key
+ * that is just created and is deleted after use.
+ *
+ * @param fn The procedure to be executed
+ * @returns The returned value of the input procedure.
+ */
+export const useEphemeralKey = async <R>(
+  fn: (ephemeralContext: CryptoContext) => Promise<R>
+): Promise<R> => {
+  // Use an ephemeral key to be destroyed after use
```

Comment:
I'm not sure this is the right approach. You may also need to access the temporary public key later in your code (for example to check the holder binding with the response). It is correct to delete it from the TEE but you need to leave the value of the public key to be able to use it later.

Reply:
As a user, you can get the public key as a result, if needed:
```ts
const [result, pubk] = await useEphemeralKey(ctx => {
  const publicKey = await ctx.getPublicKey();
  // your stuff
  return [value, publicKey];
});
```